File: SportsPlatform.Auth/SportsPlatform.Auth.Api/wwwroot/react_web/src/services/attendanceService.js
import api from './api';

// POST /clubs/:clubId/teams/:teamId/events/:eventId/attendance
export const recordAttendance = async (clubId, teamId, eventId, request) => {
  const { data } = await api.post(`/clubs/${clubId}/teams/${teamId}/events/${eventId}/attendance`, request);
  return data;
};

// GET /clubs/:clubId/teams/:teamId/events/:eventId/attendance
export const getEventAttendance = async (clubId, teamId, eventId, instanceDate) => {
  const params = {};
  if (instanceDate) params.instanceDate = instanceDate;
  const { data } = await api.get(`/clubs/${clubId}/teams/${teamId}/events/${eventId}/attendance`, { params });
  return data;
};

// PUT /clubs/:clubId/teams/:teamId/events/:eventId/attendance/:playerUserId
export const updateAttendance = async (clubId, teamId, eventId, playerUserId, request) => {
  const { data } = await api.put(`/clubs/${clubId}/teams/${teamId}/events/${eventId}/attendance/${playerUserId}`, request);
  return data;
};

// GET /clubs/:clubId/teams/:teamId/events/:eventId/attendance/me
export const getMyAttendance = async (clubId, teamId, eventId, instanceDate) => {
  const params = {};
  if (instanceDate) params.instanceDate = instanceDate;
  const { data } = await api.get(`/clubs/${clubId}/teams/${teamId}/events/${eventId}/attendance/me`, { params });
  return data;
};

File: eq-master/lib/services/attendance_service.dart
import '../models/api_models.dart';
import 'api_client.dart';

class AttendanceService {
  final ApiClient _api = ApiClient.instance;

  Future<List<AttendanceDto>> getEventAttendance(
      String clubId, String teamId, String eventId) async {
    final json =
        await _api.get('/clubs/$clubId/teams/$teamId/events/$eventId/attendance');
    return (json as List)
        .map((e) => AttendanceDto.fromJson(Map<String, dynamic>.from(e as Map)))
        .toList();
  }

  Future<dynamic> recordAttendance(
      String clubId, String teamId, String eventId, Map<String, dynamic> body) {
    return _api.post(
      '/clubs/$clubId/teams/$teamId/events/$eventId/attendance',
      body: body,
    );
  }

  Future<dynamic> updateAttendance(String clubId, String teamId, String eventId,
      String playerId, Map<String, dynamic> body) {
    return _api.put(
      '/clubs/$clubId/teams/$teamId/events/$eventId/attendance/$playerId',
      body: body,
    );
  }
}
